### PR TITLE
[IE TESTS] dynavic batch for mvn layer

### PR DESCRIPTION
**Pull request by antonzaycev96**
adding MVN layer for dynamic batch  
adding test to new infrastructure  
updating conditions for dynamic batch in MKLDNN
@AlexPeskov

### CODE REVIEW

File: inference-engine/tests/functional/plugin/shared/src/single_layer_tests/mvn.cpp [hunk 1]
```diff
@@ -54,6 +54,13 @@ void MvnLayerTest::SetUp() {
 
 TEST_P(MvnLayerTest, CompareWithRefs) {
     Run();
-};
+}
+
+TEST_P(MvnLayerTest, TestForDynamicBatch) {
+    if (targetDevice == CommonTestUtils::DEVICE_GPU || targetDevice == CommonTestUtils::DEVICE_CPU) {
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/shared/src/single_layer_tests/mvn.cpp:60]: IMO, we should create instance only with config as parameter for CPU and GPU plugins. Creation of this test case looks like extra action.

File: inference-engine/src/mkldnn_plugin/mkldnn_exec_network.cpp [hunk 2]
```diff
@@ -289,6 +289,10 @@ bool MKLDNNExecNetwork::CanProcessDynBatch(const InferenceEngine::ICNNNetwork &n
              reshapeLayer->insData[0].lock()->getTensorDesc().getDims()[0])) {
             return;
         }
+        auto MVNLayer = dynamic_cast<InferenceEngine::MVNLayer *>(layer.get());
+        if (MVNLayer && type == MVN) {
+            return;
+        }
```
Comment (AlexPeskov) [inline inference-engine/src/mkldnn_plugin/mkldnn_exec_network.cpp:295]: Transfer this check into 'if' below. just a new one line "type != MVN"

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/mvn.cpp [hunk 3]
```diff
@@ -38,13 +38,19 @@ const std::vector<double> epsilon = {
     0.000000001
 };
 
+const std::vector<std::map<std::string, std::string>> Configs = {
+        {},
+        {{CONFIG_KEY(DYN_BATCH_ENABLED), CONFIG_VALUE(YES)}}
+};
+
```
Comment (AlexPeskov) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/mvn.cpp:45]: To check dynamic batch you should pass real batch size into infer request via call of method `InferRequest::SetBatch(...)`. Not only specify config key. Please improve test body to check this functionality correctly.

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 4]
```diff
@@ -39,7 +39,12 @@ void LayerTestsCommon::Compare(const std::vector<std::uint8_t> &expected, const
     const auto actualBuffer = lockedMemory.as<const std::uint8_t *>();
 
     const auto &precision = actual->getTensorDesc().getPrecision();
-    const auto &size = actual->size();
+    auto resize = 1;
+    // With dynamic batch, you need to size
+    if (DynamicBathFlag) {
```
Comment (iefode) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:44]: `if (configuration.count(InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_ENABLED))`

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 5]
```diff
@@ -92,11 +100,14 @@ void LayerTestsCommon::Infer() {
 
     for (const auto &input : cnnNetwork.getInputsInfo()) {
         const auto &info = input.second;
-
         auto blob = GenerateInput(*info);
         inferRequest.SetBlob(info->name(), blob);
         inputs.push_back(blob);
     }
+    if (DynamicBathFlag) {
```
Comment (iefode) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:107]: `if (configuration.count(InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_ENABLED))`

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.hpp [hunk 6]
```diff
@@ -102,7 +102,8 @@ class LayerTestsCommon : public CommonTestUtils::TestsCommon {
     std::vector<InferenceEngine::Blob::Ptr> inputs;
     float threshold;
     InferenceEngine::CNNNetwork cnnNetwork;
-
+    bool DynamicBathFlag = false;  // check for dynamic batch
```
Comment (iefode) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.hpp:105]: Is it really nessesary?

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.hpp [hunk 7]
```diff
@@ -102,7 +102,7 @@ class LayerTestsCommon : public CommonTestUtils::TestsCommon {
     std::vector<InferenceEngine::Blob::Ptr> inputs;
     float threshold;
     InferenceEngine::CNNNetwork cnnNetwork;
-
+    int bathSize = 1;  // dynamic bath size ( first dimension in inputShapes > 1 !!!)
```
Comment (iefode) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.hpp:105]: Again, Is it really nessesary?

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 8]
```diff
@@ -39,7 +39,14 @@ void LayerTestsCommon::Compare(const std::vector<std::uint8_t> &expected, const
     const auto actualBuffer = lockedMemory.as<const std::uint8_t *>();
 
     const auto &precision = actual->getTensorDesc().getPrecision();
-    const auto &size = actual->size();
+    auto resize = 1;
+    auto bathSize = 1;
+    // With dynamic batch, you need to size
+    if (configuration.count(InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_ENABLED)) {
```
Comment (AlexPeskov) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:45]: Minor comment. Not fully correct check. It should be compared with "YES" value.

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 9]
```diff
@@ -39,7 +39,14 @@ void LayerTestsCommon::Compare(const std::vector<std::uint8_t> &expected, const
     const auto actualBuffer = lockedMemory.as<const std::uint8_t *>();
 
     const auto &precision = actual->getTensorDesc().getPrecision();
-    const auto &size = actual->size();
+    auto resize = 1;
+    auto bathSize = 1;
```
Comment (AlexPeskov) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:43]: is it really "bath"  :-)

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 10]
```diff
@@ -39,7 +39,14 @@ void LayerTestsCommon::Compare(const std::vector<std::uint8_t> &expected, const
...
+    // With dynamic batch, you need to size
+    if (configuration.count(InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_ENABLED)) {
+        resize = actual->getTensorDesc().getDims()[0];
+        bathSize = resize > 1 ? resize/ 2 : 1;
+    }
+    const auto &size = (actual->size() * bathSize / resize);
```
Comment (AlexPeskov) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:49]: It's better to move `actual->size() * bathSize / resize` expression into `if` statement.

like:
`
auto &size = actual->size();
if (configuration.count(InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_ENABLED)) {
         batchSize = actual->getTensorDesc().getDims()[0];
         halfBatchSize = resize > 1 ? resize/ 2 : 1;
         size = (actual->size() * halfBatchSize / batchSize)
     }
`

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 11]
```diff
@@ -92,11 +99,14 @@ void LayerTestsCommon::Infer() {
 
     for (const auto &input : cnnNetwork.getInputsInfo()) {
         const auto &info = input.second;
-
         auto blob = GenerateInput(*info);
         inferRequest.SetBlob(info->name(), blob);
         inputs.push_back(blob);
     }
+    if (configuration.count(InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_ENABLED)) {
```
Comment (AlexPeskov) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:106]: better to compare with YES

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 12]
```diff
@@ -92,11 +99,14 @@ void LayerTestsCommon::Infer() {
...
+        auto bathSize = cnnNetwork.getInputsInfo().begin()->second->getTensorDesc().getDims()[0] / 2;
```
Comment (AlexPeskov) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:107]: You are using different formula for half batch calculation. 
Previously in code : 
`batchSize = resize > 1 ? resize/ 2 : 1;`

**Pull request merged**